cmd/snap-update-ns/bootstrap_test.go

```diff
@@ -31,134 +29,50 @@ type bootstrapSuite struct{}
-		"\x00-o1\x00",
-		"arg0\x00-o1\x00",
-		"arg0\x00-o1\x00arg1\x00",
-		"arg0\x00-o1\x00-o2\x00",
-		"arg0\x00-o1\x00-o2\x00arg1\x00",
-	} {
-		result := update.FindFirstOption([]byte(str))
-		c.Assert(result, DeepEquals, &expected, Commentf("got %q", *result))
-	}
-}
-
 // Check that ValidateSnapName rejects "/" and "..".
 func (s *bootstrapSuite) TestValidateSnapName(c *C) {
 	c.Assert(update.ValidateSnapName("hello-world"), Equals, 0)
 	c.Assert(update.ValidateSnapName("hello/world"), Equals, -1)
 	c.Assert(update.ValidateSnapName("hello..world"), Equals, -1)
 	c.Assert(update.ValidateSnapName("INVALID"), Equals, -1)
 	c.Assert(update.ValidateSnapName("-invalid"), Equals, -1)
+	c.Assert(update.ValidateSnapName(""), Equals, -1)
 }
 
 // Test various cases of command line handling.
 func (s *bootstrapSuite) TestProcessArguments(c *C) {
 	cases := []struct {
-		cmdline     string
+		cmdline     []string
 		snapName    string
 		shouldSetNs bool
 		errPattern  string
 	}{
 		// Corrupted buffer is dealt with.
-		{"", "", false, "argv0 is corrupted"},
+		{[]string{}, "", false, "argv0 is corrupted"},
 		// When testing real bootstrap is identified and disabled.
-		{"argv0.test\x00", "", false, "bootstrap is not enabled while testing"},
+		{[]string{"argv0.test"}, "", false, "bootstrap is not enabled while testing"},
 		// Snap name is mandatory.
-		{"argv0\x00", "", false, "snap name not provided"},
-		{"argv0\x00\x00", "", false, "snap name not provided"},
+		{[]string{"argv0"}, "", false, "snap name not provided"},
 		// Snap name is parsed correctly.
-		{"argv0\x00snapname\x00", "snapname", true, ""},
+		{[]string{"argv0", "snapname"}, "snapname", true, ""},
+		// Onlye one snap name is allowed.
+		{[]string{"argv0", "snapone", "snaptwo"}, "", false, "too many positional arguments"},
 		// Snap name is validated correctly.
-		{"argv0\x00in--valid\x00", "", false, "snap name cannot contain two consecutive dashes"},
-		{"argv0\x00invalid-\x00", "", false, "snap name cannot end with a dash"},
-		{"argv0\x00@invalid\x00", "", false, "snap name must use lower case letters, digits or dashes"},
-		{"argv0\x00INVALID\x00", "", false, "snap name must use lower case letters, digits or dashes"},
+		{[]string{"argv0", ""}, "", false, "snap name must contain at least one letter"},
+		{[]string{"argv0", "in--valid"}, "", false, "snap name cannot contain two consecutive dashes"},
+		{[]string{"argv0", "invalid-"}, "", false, "snap name cannot end with a dash"},
+		{[]string{"argv0", "@invalid"}, "", false, "snap name must use lower case letters, digits or dashes"},
+		{[]string{"argv0", "INVALID"}, "", false, "snap name must use lower case letters, digits or dashes"},
 		// The option --from-snap-confine disables setns.
-		{"argv0\x00--from-snap-confine\x00snapname\x00", "snapname", false, ""},
+		{[]string{"argv0", "--from-snap-confine", "snapname"}, "snapname", false, ""},
+		{[]string{"argv0", "snapname", "--from-snap-confine"}, "snapname", false, ""},
```

Comment:
This is different from the original design (<name> followed by <option>), but I think that is ok.

Reply:
The existing code seems to allow either order (as does the Go argument parsing code), so I went ahead and added the extra test.  We could turn it into an error, but that'd likely make the code a bit more complicated.

Reply:
Right, that's precisely why I don't think it is important (adding code complexity). IMO, don't worry about it.